tests/storage/restore.go

```diff
@@ -307,28 +309,29 @@ var _ = SIGDescribe("VirtualMachineRestore Tests", func() {
 			It("[test_id:5256]should successfully restore", func() {
 				var origSpec *v1.VirtualMachineSpec
 
-				Eventually(func() bool {
-					var updatedVM *v1.VirtualMachine
+				By("Wait for snapshot to be finished")
+				Eventually(func() *string {
 					vm, err = virtClient.VirtualMachine(vm.Namespace).Get(vm.Name, &metav1.GetOptions{})
 					Expect(err).ToNot(HaveOccurred())
 
-					if vm.Status.SnapshotInProgress != nil {
-						return false
-					}
+					return vm.Status.SnapshotInProgress
+				}, 180*time.Second, time.Second).Should(BeNil())
 
-					origSpec = vm.Spec.DeepCopy()
-					Expect(origSpec.Template.Spec.Domain.Resources.Requests[corev1.ResourceMemory]).To(Equal(resource.MustParse("128Mi")))
+				initialRequestedMemory := resource.MustParse("128Mi")
+				Expect(vm.Spec.Template.Spec.Domain.Resources.Requests[corev1.ResourceMemory]).To(Equal(initialRequestedMemory))
 
-					vm.Spec.Template.Spec.Domain.Resources.Requests[corev1.ResourceMemory] = resource.MustParse("256Mi")
-					updatedVM, err = virtClient.VirtualMachine(vm.Namespace).Update(vm)
-					if errors.IsConflict(err) {
-						return false
+				origSpec = vm.Spec.DeepCopy()
+
+				err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+					vm, err = virtClient.VirtualMachine(vm.Namespace).Get(vm.Name, &metav1.GetOptions{})
+					if err != nil {
+						return err
 					}
-					vm = updatedVM
-					Expect(err).ToNot(HaveOccurred())
-					Expect(vm.Spec.Template.Spec.Domain.Resources.Requests[corev1.ResourceMemory]).To(Equal(resource.MustParse("256Mi")))
-					return true
-				}, 180*time.Second, time.Second).Should(BeTrue())
+					increasedRequestedMemory := resource.MustParse("256Mi")
+					vm.Spec.Template.Spec.Domain.Resources.Requests[corev1.ResourceMemory] = increasedRequestedMemory
+					vm, err = virtClient.VirtualMachine(vm.Namespace).Update(vm)
```

Comment:
I believe we should either not wrap this in retry or directly convert it to patch.

Reply:
This commit is an attempt to organize the code in a cleaner way, i.e. refactor, without changing any logic.
It allows the next commit to perform the move from update to patch in a manner that is visible and clear.

The usage of the retry on the update is basically correct, and I prefer to see the commit logically correct, even if it is an intermediate step.